### PR TITLE
Move nci_usage to dataset

### LINKED ISSUE #14: Design of tools within ncimonitor
@ScottWales @aidanheerdegen When building the tool for reporting I found some useful stuff in both ncimonitor and nciusage. But since nciusage main class is independent from ncimonitor main class, mixing both was annoying. For example, ncimonitor uses dataset to connect to the db but nciusage uses sqlite3. 
If we want to build more tools on the db, would it make sense to rethink the design? For example, a simple superclass with __init__ only to connect to the db + a few informative attributes and sub-classes of this used in tools? So we ensure some coherence in objects? Or am I just not experienced enough in Python?

### PR DESCRIPTION
**Pull request by ScottWales**
Fixes #14

### CODE REVIEW

File: ncimonitor/UsageDataset.py [hunk 1]
```diff
@@ -314,3 +314,39 @@ def getsystemstorage(self, systemname, storagepoint, year, quarter):
...
+        or 'inodes') on ``storagepoint`` at ``date``
+
+        Due to the NCI scanning process ``date`` should be at least one day in
+        the past
+
+        Returns columns (measure, username, fullname)
+        """
+
+        if storagepoint not in ['short', 'gdata']:
+            raise Exception(f"Unexpected storagepoint '{storagepoint}'")
```
Comment (aidanheerdegen) [inline ncimonitor/UsageDataset.py]: Change to format or drop python 2 support. Your choice. I'm happy with the latter. We don't intend to use it with python 2 in the future.

**Comment by ScottWales** [comment]
Dropped python 2 support

File: ncimonitor/UsageDataset.py [hunk 2]
```diff
@@ -314,3 +314,39 @@ def getsystemstorage(self, systemname, storagepoint, year, quarter):
...
+
+        if measure not in ['size', 'inodes']:
+            raise Exception(f"Unexpected measure '{measure}'")
+
+        if date is None:
+            # Default to one day in the past
+            date = datetime.date.today() - datetime.timedelta(days=1)
+
+        q = self.db.query(f"""
```
Comment (aidanheerdegen) [inline ncimonitor/UsageDataset.py]: I was sort of hoping to converge on minimising SQL queries and use the pandas dataframes as a unit of interchange. For example, the same functionality can be done by calling 

```
(Pdf) dp = db.getstorage(year, quarter, storagept=storagept, datafield=datafield)
(Pdb) p dp.ix[-1].sort_values().tail(5)
Name
G**** L**** (gl****)      6.536940e+12
N**** C**** (nc****)      7.505455e+12
A**** H**** (am****)      8.724162e+12
P**** S**** (pa****)      2.006201e+13
A**** K**** (ae****)      2.430116e+13
```

Comment (aidanheerdegen) [reply ncimonitor/UsageDataset.py]: The rationale is to make it easier for others to add functionality without SQL knowledge, as they can follow examples like above.

**Pull request merged**